File: hackerearth/maxor.hs

import Data.Bits

-- solve :: (Num a) => [a] -> [a]

solve [] = 1 : replicate 127 0

solve (x:xs) = [(count !! i) + (count !! (i `xor` x)) |  i <- [0..127]]
                where count = solve xs

-- ans :: (Num a) => [a] -> b

nC2 (x:xs) = sum $ map (\x -> (x * (x - 1)) / 2) $ filter (> 1) xs

main = do
    n <- getLine
    numStrs <- sequence $ replicate ((read n)::Int) getLine
    let numbers = map (read :: String -> Int) numStrs
    print $ truncate $ nC2 $ solve $ numbers

File: hackerearth/maxor.cpp
#include <bits/stdc++.h>

using namespace std;




int main() {

    vector<long long> subset_cnt(128);
    subset_cnt[0] = 1;

    long long n;
    cin >> n;

    int a;

    while(n--) {

        vector<long long> new_subset_cnt = subset_cnt;

        cin >> a;

        for(int i = 0; i < 128; i++)
            if(subset_cnt[i])   new_subset_cnt[i ^ a] += subset_cnt[i];

        subset_cnt = new_subset_cnt;
    }


    long long ans = 0;

    for(int i = 1; i < 128; i++) {
        if(subset_cnt[i] > 1)
            ans += (subset_cnt[i] * (subset_cnt[i] - 1)) / 2;
    }

    cout << ans << endl;
}
